### PR TITLE
BAU: Nested set finale

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

BAU

### What?

I have added/removed/altered:

- [x] Replaces the goods nomenclature mapper with nested set in the headings, chapter and commodity endpoints of v1
- [x] Fixes parts of the v2 endpoints that were missed in previous migrations
- [x] Removes a whole bunch of now redundant functionality
- [x] Removes a whole bunch of now redundant tests

### Why?

I am doing this because:

- This is required to enable a huge cleanup of the code base

### CODE REVIEW

File: app/presenters/api/v2/commodities/commodity_presenter.rb [hunk 1]
```diff
@@ -65,8 +65,14 @@ def zero_mfn_duty?
           third_country_measures.size.positive? && third_country_measures.all?(&:zero_mfn?)
         end
 
+        def basic_duty_rate
+          if third_country_measures.one?
+            third_country_measures.first&.formatted_duty_expression
+          end
+        end
```
Comment (alexdesi) [inline app/presenters/api/v2/commodities/commodity_presenter.rb:72]: so if there are > 1 `third_country_measures` the basic_duty_rate is `nil?`

File: app/presenters/api/v2/headings/declarable_heading_presenter.rb [hunk 2]
```diff
@@ -59,6 +59,12 @@ def entry_price_system?
           import_measures.any?(&:entry_price_system?)
         end
 
+        def basic_duty_rate
+          if third_country_measures.one?
+            third_country_measures.first&.formatted_duty_expression
+          end
+        end
```
Comment (alexdesi) [inline app/presenters/api/v2/headings/declarable_heading_presenter.rb:66]: It's not a big issue, but this method is duplicated.

**Review by alexdesi — COMMENTED** [review]
Really nice cleaning here! I can see the effort you've put in here! 🦸 
I didn't understand everything, it'd take 2 hours to review it properly,
but I've put some hopefully useful comments.

File: app/controllers/api/v1/headings_controller.rb [hunk 3]
```diff
@@ -55,26 +93,39 @@ def changes
       end
 
       def tree
-        @heading = Heading.actual
-                          .non_grouping
-                          .where(goods_nomenclatures__goods_nomenclature_item_id: params[:id])
-                          .take
-        @commodities = GoodsNomenclatureMapper.new(@heading.commodities_dataset.eager(:goods_nomenclature_indents, :goods_nomenclature_descriptions).all).all
+        @heading = non_declarable_heading
```
[the remark this answers is not in the document]
Comment (willfish) [reply app/controllers/api/v1/headings_controller.rb:96]: `tree` is an endpoint, yeah

File: app/presenters/api/v2/commodities/commodity_presenter.rb (hunk 1 above)
Comment (willfish) [reply app/presenters/api/v2/commodities/commodity_presenter.rb:72]: That's the situation that I've copied, yeah

File: app/presenters/api/v2/headings/declarable_heading_presenter.rb (hunk 2 above)
Comment (willfish) [reply app/presenters/api/v2/headings/declarable_heading_presenter.rb:66]: Yeah, I'm not even convinced we use these anywhere

Comment (willfish) [reply app/presenters/api/v2/headings/declarable_heading_presenter.rb:66]: And by use anywhere I mean the frontend declares, but does not implement them. We've essentially replaced this with the import trade summary you built in 2021

**Review by maverickvi — APPROVED** [review]

**Pull request merged**